### PR TITLE
Adding h5coro package

### PR DESCRIPTION
**Pull request by jpswinski**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jpswinski** [comment]
@conda-forge/help-python, ready for review!

File: recipes/h5coro/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/h5coro-{{ version }}.tar.gz
+  sha256: 9665cea002b10e3cb262f92d33768a0a116e3233fcc0848db8f81f68db626151
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/h5coro/meta.yaml:13]: The scripts entry in this package, which prevents it from being noarch, is empty. See https://github.com/ICESat2-SlideRule/h5coro/blob/v0.0.4/setup.py#L19

So it is safe to make it noarch. I would also report that upstream.

```suggestion
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/h5coro/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/h5coro/meta.yaml:22]:
```suggestion
    - python >=3.6
    - pip
  run:
    - python >=3.6
```

Comment (ocefpaf) [reply recipes/h5coro/meta.yaml:22]: Please check upstream the min Python version. I used 3.6 as an example here.

**Pull request merged**